com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/model/MRestView.java

```diff
@@ -0,0 +1,258 @@
+		super(ctx, REST_View_UU, trxName, virtualColumns);
+	}
+
+	public MRestView(Properties ctx, ResultSet rs, String trxName) {
+		super(ctx, rs, trxName);
+	}
+	
+	public MRestView(MRestView copy) {
+		this(Env.getCtx(), copy);
+	}
+	
+	public MRestView(Properties ctx, MRestView copy) {
+		this(ctx, copy, (String) null);
+	}
+	
+	public MRestView(Properties ctx, MRestView copy, String trxName) {
+		//-1 to avoid infinite loop
+		this(ctx, -1, trxName);
+		copyPO(copy);
+	}
+	
+	@Override
+	protected boolean afterSave(boolean newRecord, boolean success) {
+		if (newRecord && success) {
+			copyColumns();
+		} else if (success) {
+			if (is_ValueChanged(COLUMNNAME_AD_Table_ID)) {
+				getColumns(true);
+				for(MRestViewColumn column : columns) {
+					column.deleteEx(true, get_TrxName());
+				}
+				copyColumns();
+				columns = null;
+				getRelatedViews(true);
+				for(MRestViewRelated related : relateds) {
+					related.deleteEx(true, get_TrxName());
+				}
+				relateds = null;
+			}
+		}
+		return success;
+	}
+
+	/**
+	 * Copy all columns from table
+	 */
+	private void copyColumns() {
+		//auto populate view columns
+		MTable table = MTable.get(getAD_Table_ID());
+		MColumn[] columns = table.getColumns(false);
+		String keyColumn = table.getKeyColumns() != null && table.getKeyColumns().length == 1 ? table.getKeyColumns()[0] : "";
+		String uidColumn = PO.getUUIDColumnName(table.getTableName());
+		for(MColumn column : columns) {
+			if (column.getColumnName().equals(keyColumn))
+				continue;
+			else if (column.getColumnName().equals(uidColumn))
+				continue;
+			MRestViewColumn restViewColumn = new MRestViewColumn(Env.getCtx(), 0, get_TrxName());
+			restViewColumn.setREST_View_ID(getREST_View_ID());
+			restViewColumn.setAD_Column_ID(column.getAD_Column_ID());
+			restViewColumn.setName(TypeConverterUtils.toPropertyName(column.getColumnName()));
+			restViewColumn.saveEx();
+		}
+	}
+
+	/**
+	 * Get view by name
+	 * @param name
+	 * @return MRestView
+	 */
+	public static MRestView get(String name) {
+		MRestView view = s_cache.get(Env.getCtx(), name, e -> new MRestView(Env.getCtx(), e));
+		if (view != null)
+			return view;
+		
+		Query query = new Query(Env.getCtx(), Table_Name, "Name=?", null);
+		view = query.setParameters(name).first();
+		if (view != null) {
+			s_cache.put (name, view, e -> new MRestView(Env.getCtx(), e));
+		}
+		return view;
+	}
+	
+	/**
+	 * Get view columns
+	 * @return view columns
+	 */
+	public MRestViewColumn[] getColumns() {
+		return getColumns(false);
+	}
+	
+	/**
+	 * Get view columns
+	 * @param refresh
+	 * @return view columns
+	 */
+	public MRestViewColumn[] getColumns(boolean refresh) {
+		if (columns != null && !refresh)
+			return columns;
+		
+		Query query = new Query(Env.getCtx(), MRestViewColumn.Table_Name, "REST_View_ID=?", null);
+		List<MRestViewColumn> columnList = query.setOnlyActiveRecords(true)
```

Comment:
Because of the setOnlyActiveRecords when recreating the columns after changing the table, all those columns that were hidden remained there and can be activated pointing to the wrong table.

I created a view for locations, then used the select active columns tab to hide some columns, then changed the table to m_product_category, so I can still activate some locations columns

![image](https://github.com/user-attachments/assets/d3567916-6589-48c8-8dc4-1837bcdd9452)